components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/deployment/OsgiRestApiDeployer.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.uuf.internal.deployment;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.carbon.uuf.api.RestApi;
+import org.wso2.carbon.uuf.internal.exception.RestApiDeploymentException;
+import org.wso2.msf4j.Microservice;
+
+import java.util.Dictionary;
+import java.util.Hashtable;
+
+/**
+ * Deploys UI-specific REST APIs as OSGi services.
+ * This implementation depends on the OSGi mode of MSF4J. REST APIs that are deployed should be {@link Microservice}s.
+ *
+ * @since 1.0.0
+ */
+@Component(name = "org.wso2.carbon.uuf.internal.deployment.OsgiRestApiDeployer",
+           service = RestApiDeployer.class,
+           immediate = true,
+           property = {
+                   "componentName=wso2-uuf-OSGi-REST-API-deployer"
+           }
+)
+public class OsgiRestApiDeployer implements RestApiDeployer {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OsgiRestApiDeployer.class);
+    private BundleContext bundleContext;
+
+    @Activate
+    protected void activate(BundleContext bundleContext) {
+        this.bundleContext = bundleContext;
+        LOGGER.debug("OsgiRestApiDeployer activated.");
+    }
+
+    @Deactivate
+    protected void deactivate(BundleContext bundleContext) {
+        this.bundleContext = null;
+        LOGGER.debug("OsgiRestApiDeployer deactivated.");
+    }
+
+    /**
+     * {@inheritDoc}
```

Comment:
AFAIR, this tag causes issues in the build machine (maven javadoc plugin). It was a known issue at that time which might have fixed by now. Better to check it.